core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -1616,7 +1616,7 @@ public boolean isConsistent() {
     public void isConsistentOrThrow() throws IllegalStateException {
         lock.lock();
         try {
-            Set<Transaction> transactions = getTransactions(true);
+            Collection<Transaction> transactions = this.transactions.values();
```

Comment:
This is safe because we use the same lock as in getTransactions() and we don't change the transactions.

This is performant, as we don't need the `Set` property. Only iteration.